api-controllers/ServiceFabrikApiController.js

```diff
@@ -147,14 +147,16 @@ class ServiceFabrikApiController extends FabrikBaseController {
         resourceId: req.params.instance_id
       })
       .then(resource => {
+        let resourceState = _.isEmpty(_.get(resource, 'status.appliedOptions')) ? _.get(resource, 'spec.options') :
+          _.get(resource, 'status.appliedOptions');
         if (!_.get(req, 'body.context')) {
-          _.set(req, 'body.context', _.get(resource, 'spec.options.context'));
+          _.set(req, 'body.context', _.get(resourceState, 'context'));
```

Comment:
You are getting the options, so let's not call it state, let's call it resourceOptions.

Reply:
Agreed. Will modify.